### PR TITLE
Allow retries for subprocess commands

### PR DESCRIPTION
**Pull request by michael-sbarra-apexai**

### CODE REVIEW

File: repos/config/detail/lock_repos.py [hunk 1]
```diff
@@ -104,25 +104,46 @@ def extract_archive_root_folder(path, origin):
...
+    max_retries = 2
+    for i in range(max_retries + 1):
+        with tempfile.TemporaryDirectory() as tempdir:
+            try:
+                result = subprocess.run(
+                    ["git", "clone", url, "--no-checkout", tempdir, "--depth", "1",
+                    "--branch", version, "--bare", "-q"],
+                    capture_output = True,
+                    encoding='utf8',
+                    timeout=10
```
Comment (michael-sbarra-apexai) [inline repos/config/detail/lock_repos.py:116]: This timeout might be too small.

Comment (michael-sbarra-apexai) [reply repos/config/detail/lock_repos.py:116]: Either the timeout should be increased, or the max_retries.

Comment (michael-sbarra-apexai) [reply repos/config/detail/lock_repos.py]: Increased to `20s`

**Review by evan-flynn-apexai — APPROVED** [review]

**Comment by kilian-funk** [comment]
@michael-sbarra-apexai What is the purpose of this change? If the timeout is too short, can you not simply increase the timeout instead of adding all of this complexity to the code?

**Comment by michael-sbarra-apexai** [comment]
> @michael-sbarra-apexai What is the purpose of this change? If the timeout is too short, can you not simply increase the timeout instead of adding all of this complexity to the code?

@kilian-funk See #21. If some brief network interruption occurred, the whole file builder would be lost resulting in the error shown in the issue. I also found sometimes the `git ...` command would hang and not terminate, which is why I added the timeout. This now adds a `max_retries = 3` with a timeout of `20s` (for `git clone`), so the file build will not fail on some brief network issue. In additional testing, I found that timeout of `5s` should be sufficient.

**Pull request merged**